knora/dsplib/models/user.py

```diff
@@ -675,41 +675,53 @@ def getAllUsers(con: Connection) -> list[Any]:
         result = con.get(User.ROUTE)
         if 'users' not in result:
             raise BaseError("Request got no users!")
-        return list(map(lambda a: User.fromJsonObj(con, a), result['users']))
+        return [User.fromJsonObj(con, a) for a in result['users']]
 
     @staticmethod
-    def getAllUsersForProject(con: Connection, proj_shortcode: str) -> list[Any]:
+    def getAllUsersForProject(con: Connection, proj_shortcode: str) -> Optional[list[User]]:
         """
-        Get a list of all users that belong to a project(static method)
+        Get a list of all users that belong to a project (static method)
 
         :param con: Connection instance
         :project_shortcode: Shortcode of the project
         :return: List of users belonging to that project
         """
-
-        result = con.get(User.ROUTE)
-        if 'users' not in result:
-            raise BaseError("Request got no users!")
-        all_users = result["users"]
-        project_users = []
-        for user in all_users:
-            project_list = con.get(
-                User.IRI + urllib.parse.quote_plus(user["id"], safe='') + '/project-memberships')
-            project = project_list["projects"]
-            for proj in project:
-                if proj["id"] == "http://rdfh.ch/projects/" + proj_shortcode:
-                    project_users.append(user)
-        return list(map(lambda a: User.fromJsonObj(con, a), project_users))
-
-    def createDefinitionFileObj(self):
-        user = {
+        members = con.get(f'/admin/projects/shortcode/{proj_shortcode}/members')
+        if members is None or len(members) < 1:
```

Comment:
this can be simplified to `if not members:` (both `None` and empty lists/dicts/sets evaluate as false in an if-clause)